### ISSUE TITLE
Migrate FSharpQA suite to xUnit suite(s)

### ISSUE THREAD

**Issue by vzarytovskii** [dotnet/fsharp#10001]
We want to migrate FSharpQA tests from a custom runner/solution to a standard dotnet test project(s).
* Tests have to be migrated to one of following:
  * FSharp.Compiler.ComponentTests - if it's a "blackbox"-style test for the compiler, when we want to check compiler output itself, e.g. IL generation, compiler errors and warnings tests, conformance tests, etc.
  * FSharp.Compiler.UnitTests - if it's a test for a compiler "API".
  * FSharp.Core.UnitTests - if it's a test for library functions.

Prerequisite: https://github.com/dotnet/fsharp/issues/10000

**Comment by dsyme** [comment]
> FSharp.Compiler.ComponentTests - if it's a "blackbox"-style test for the compiler, when we want to check compiler output itself, e.g. IL generation, compiler errors and warnings tests, conformance tests, etc.

> FSharp.Compiler.UnitTests - if it's a test for a compiler "API".

I find these names a bit confusing `ComponentTests` sounds like `Unit Tests` to me.

Maybe aim for this:

- FSharp.Core.UnitTests // tests for FSharp.Core

- FSharp.Compiler.UnitTests // unit tests for FSharp.Compiler and FSharp.Compiler.Service

- FSharp.Compiler.UsageTests // tests actual usage of language and tools 

- VisualFSharp....UnitTests // tests for anything involving Visual Studio, Roslyn

So basically rename "ComponentTests" to "UsageTests"

**Comment by vzarytovskii** [comment]
> I find these names a bit confusing `ComponentTests` sounds like `Unit Tests` to me.

My idea was that we test compiler as an "external", separate "component", hence the category "ComponentTests".

We already have this project, renaming it should be easy though.

**Comment by cartermp** [comment]
Should "UsageTests" just be "LanguageTests"? Since that's basically what they are. Tooling tests would go under relevant tooling subfolders anyways

**Comment by vzarytovskii** [comment]
> Should "UsageTests" just be "LanguageTests"? Since that's basically what they are. Tooling tests would go under relevant tooling subfolders anyways

Keep in mind, that currently, there are not only "Language" tests there, but also stuff like IL verification, Interop, I guess if it falls under "LanguageTests", then I guess, we can have it as "LanguageTests"

**Comment by dsyme** [comment]
Now covered by https://github.com/dotnet/fsharp/issues/12823
